src/dialects/mariadb/query.js

```diff
@@ -184,7 +185,7 @@ class Query extends AbstractQuery {
       if (modelField.type instanceof DataTypes.JSON) {
         // Value is returned as String, not JSON
         rows = rows.map(row => {
-          if (row[modelField.fieldName] && typeof row[modelField.fieldName] === 'string') {
+          if (row[modelField.fieldName] && typeof row[modelField.fieldName] === 'string' && !this.connection.info.hasMinVersion(10, 5, 2)) {
```

Comment:
Could we add a link to the docs that describe this behavior?

Reply:
I'm not sure if you mean to add documentation to sequelize indicating that JSON fields for mariadb < 10.5.2 will be returned as strings and not as data (unlike other db's) or you mean to add a link in the code that indicates a reason for the mariadb version check: this is not clearly specified anywhere but from testing it turns out that it is, the reference links are anyway:
https://github.com/mariadb-corporation/mariadb-connector-nodejs/issues/136 and https://jira.mariadb.org/browse/MDEV-17832

Reply:
also this PR points to v6 but I think it's the same with v7

Reply:
Yeah a link that references the Maria documentation and explains that the parsing happens automatically from a specific version on

Reply:
added a comment.
also note that version check is already used with mariadb-connector-nodejs to test this same thing:

https://github.com/mariadb-corporation/mariadb-connector-nodejs/blob/59da962ce4ae6338bb14ed8854a70526b429911f/test/integration/datatype/test-json.js#L101

https://github.com/mariadb-corporation/mariadb-connector-nodejs/blob/59da962ce4ae6338bb14ed8854a70526b429911f/types/index.d.ts#L108